service/controller/clusterapi/v17/resources/clusterid/create.go

```diff
@@ -2,8 +2,52 @@ package clusterid
 
 import (
 	"context"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/controller/context/reconciliationcanceledcontext"
+
+	"github.com/giantswarm/cluster-operator/service/controller/clusterapi/v17/key"
 )
 
 func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	cr, err := key.ToCluster(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", "ensuring cluster status has cluster ID")
+
+	status := r.commonClusterStatusAccessor.GetCommonClusterStatus(cr)
+
+	if status.ID != "" {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "ensured cluster status has cluster ID")
+		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+
+		return nil
+	}
+
+	clusterID := key.ClusterID(&cr)
+
+	if clusterID == "" {
+		return microerror.Maskf(executionFailedError, "cluster ID missing from CR")
+	}
+
+	status.ID = clusterID
+
+	updatedCR := r.commonClusterStatusAccessor.SetCommonClusterStatus(cr, status)
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", "updating cluster status")
+
+	_, err = r.cmaClient.ClusterV1alpha1().Clusters(cr.Namespace).UpdateStatus(&updatedCR)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", "updated cluster status")
+	r.logger.LogCtx(ctx, "level", "debug", "message", "ensured cluster status has cluster ID")
+	r.logger.LogCtx(ctx, "level", "debug", "message", "canceling reconciliation")
```

Comment:
I miss a lot of context here but why are we canceling the resource?

I think comments for that would be good.

Reply:
I added a comment to explain this. It's because further resources require cluster ID to be present and when we just wrote it there, the object used for further resource execution would be lacking it so it makes sense to start from the beginning :slightly_smiling_face:

Reply:
Great thanks for adding! ❤️